integrations/jira/jira_test.go

```diff
@@ -0,0 +1,109 @@
+package jira
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// createMockServer returns a mock server with custom headers and response body for tests.
+func createMockServer(id string, header int) *httptest.Server {
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		r := map[string]string{
+			"key": string(id),
+		}
+
+		b, _ := json.Marshal(r)
+		rw.WriteHeader(header)
+		rw.Write(b)
+	}))
+
+	return server
+}
+
+// createMockClient returns a mock Client for tests.
+func createMockClient() *Client {
+	c := &Client{
+		Client: http.DefaultClient,
+		config: &Config{
+			URI:        "https://jira.atlassian.net",
+			Token:      "abc",
+			Username:   "test",
+			APIVersion: "latest",
+		},
+	}
+
+	return c
+}
+
+// createMockIssue returns a mock Issue for tests.
+func createMockIssue() Issue {
+	i := Issue{
+		Key: "1",
+		Fields: &IssueFields{
+			Project: &Project{
+				Key: "Test",
+			},
+			Summary:     "Test Issue",
+			Description: "Test Description",
+		},
+	}
+
+	return i
+}
+
+func TestNewClient(t *testing.T) {
+	c := createMockClient()
+
+	t.Run("config URI should be same as the value passed to NewClient()", func(t *testing.T) {
+		got := NewClient("https://jira.atlassian.net", "test", "abc", "latest")
+		if got.config.URI != "https://jira.atlassian.net" {
```

Comment:
I'll rewrite this as: `got.config.URI != c.config.URI` in all tests; any other nits?